src/tests/illumidesk/apis/test_jupyterhub_api.py

```diff
@@ -59,6 +60,27 @@ def test_initializer_sets_headers_to_make_requests(jupyterhub_api_environ):
     assert sut.default_headers['Authorization'] == f'token {os.environ.get("JUPYTERHUB_API_TOKEN")}'
 
 
+@pytest.mark.asyncio
+async def test_request_raises_value_error_with_endpoint_empty(jupyterhub_api_environ):
+    """
+    Does the _request method accespt an missing endpoint argument?
+    """
+    with pytest.raises(ValueError):
+        await JupyterHubAPI()._request(endpoint='')
+
+
+def test_request_builds_correct_url_variable(jupyterhub_api_environ):
+    """
+    Does the _request build the correct url based on the api url attribute and the
+    endpoint argument?
+    """
+    endpoint = 'users'
+    api_root_url = 'https://localhost/hub/api'
+    actual = f'{api_root_url}/{endpoint}'
+    expected = 'https://localhost/hub/api/users'
+    assert expected == actual
```

Comment:
hm i think it's incomplete, right? why not use the wrapper?

Reply:
This is just a simple test to ensure the endpoint string is built correctly regardless of the client used to call this endpoint.